### PR TITLE
Basic statistics sparse fix and adding test

### PR DESCRIPTION
**Pull request by md-shafiul-alam**
## Description

_Add a comprehensive description of proposed changes_

_List associated issue number(s) if exist(s): #6 (for example)_

_Documentation PR (if needed): #1340 (for example)_

_Benchmarks PR (if needed): https://github.com/IntelPython/scikit-learn_bench/pull/155 (for example)_

---

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] I have run it locally and tested the changes extensively.
- [ ] All CI jobs are green or I have provided justification why they aren't.
- [ ] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [ ] I have provided justification why performance has changed or why changes are not expected.
- [ ] I have provided justification why quality metrics have changed or why changes are not expected.
- [ ] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

File: sklearnex/basic_statistics/tests/test_basic_statistics.py [hunk 1]
```diff
@@ -178,6 +181,55 @@ def test_multiple_options_on_random_data(
...
+        n_samples=row_count, n_features=column_count, random_state=random_state
+    )
+    density = 0.05
+    X_sparse = csr_matrix(X * (np.random.rand(*X.shape) < density))
+    X_dense = X_sparse.toarray()
+
+    if weighted:
+        weights = gen.uniform(low=-0.5, high=1.0, size=row_count)
+        weights = weights.astype(dtype=dtype)
+    basicstat = BasicStatistics(result_options=["mean", "max", "sum"])
```
Comment (Vika-F) [inline sklearnex/basic_statistics/tests/test_basic_statistics.py:205]: According to onedal tests, need to exclude "max" at it contains bugs:
https://github.com/intel/scikit-learn-intelex/blob/main/onedal/basic_statistics/tests/test_basic_statistics.py#L273

Comment (icfaust) [reply sklearnex/basic_statistics/tests/test_basic_statistics.py:205]: > According to onedal tests, need to exclude "max" at it contains bugs: https://github.com/intel/scikit-learn-intelex/blob/main/onedal/basic_statistics/tests/test_basic_statistics.py#L273

@olegkkruglov please message out a link to the ticket associated with this error (just to make sure it wasn't lost)

Comment (icfaust) [reply sklearnex/basic_statistics/tests/test_basic_statistics.py]: Sorry, my mistake, I didn't dig deep enough in the git blame.  Turns out it was introduced here https://github.com/intel/scikit-learn-intelex/pull/1846 by @Vika-F .  Do you know if there was any follow-up work after #1846 on the max issues/ any memory on what was going on?

Comment (md-shafiul-alam) [reply sklearnex/basic_statistics/tests/test_basic_statistics.py]: I observed that issue as well, temporarily removed the "max" in tests.

File: sklearnex/basic_statistics/tests/test_basic_statistics.py [hunk 2]
```diff
@@ -178,6 +181,53 @@ def test_multiple_options_on_random_data(
     assert_allclose(gtr_sum, res_sum, atol=tol)
 
 
+@pytest.mark.parametrize("queue", get_queues())
```
Comment (samir-nasibli) [inline sklearnex/basic_statistics/tests/test_basic_statistics.py:184]: please use `_get_dataframes_and_queues` instead

Comment (md-shafiul-alam) [reply sklearnex/basic_statistics/tests/test_basic_statistics.py:184]: Sparse data can't work with dataframes

**Comment by Vika-F** [comment]
Moved the development to #2253
